### ISSUE TITLE
Random crashes in latest nightly

### ISSUE THREAD

**Issue by Dutchman101** [multitheftauto/mtasa-blue#1810]
**Describe the bug**
The latest nightly is unstable, due to a random crash being introduced, presumably by 2f4c0ba. It happens during mid-gameplay, and no repro is known. In fact, facing several bikes (including ones with modified handling) during your session is possible before crashing.

For pointing to PR #1701 and the reason I referred to bikes, see stacktrace further down below.

Here are 2 example crash dumps: [https://upload.mtasa.com/u/782515977/dumps.zip_](https://upload.mtasa.com/u/782515977/dumps.zip_)

```
** 
Version = 1.5.8-release-20717.0.000
Time = Sun Nov  8 18:51:48 2020
Module = C:\Program Files (x86)\MTA San Andreas 1.5\mta\game_sa.dll
Code = 0xC0000005
Offset = 0x0000BCD3

EAX=66722CB8  EBX=0177F924  ECX=00000000  EDX=00000004  ESI=00000007
EDI=1C1CE638  EBP=0177F780  ESP=0177F77C  EIP=6668BCD3  FLG=00010206
CS=0023   DS=002B  SS=002B  ES=002B   FS=0053  GS=002B
```

<details><summary>Dumptrace 1:</summary>

```
CONTEXT:  (.ecxr)
eax=66722cb8 ebx=0177f924 ecx=00000000 edx=00000004 esi=00000007 edi=1c1ce638
eip=6668bcd3 esp=0177f77c ebp=0177f780 iopl=0         nv up ei pl nz na pe nc
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00010206
game_sa!CBikeSA::RecalculateBikeHandling+0x3:
6668bcd3 8b01            mov     eax,dword ptr [ecx]  ds:002b:00000000=????????
Resetting default scope

EXCEPTION_RECORD:  (.exr -1)
ExceptionAddress: 6668bcd3 (game_sa!CBikeSA::RecalculateBikeHandling+0x00000003)
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000000
NumberParameters: 2
   Parameter[0]: 00000000
   Parameter[1]: 00000000
Attempt to read from address 00000000

PROCESS_NAME:  gta_sa.exe

READ_ADDRESS:  00000000 

ERROR_CODE: (NTSTATUS) 0xc0000005 - The instruction at 0x%p referenced memory at 0x%p. The memory could not be %s.

EXCEPTION_CODE_STR:  c0000005

EXCEPTION_PARAMETER1:  00000000

EXCEPTION_PARAMETER2:  00000000

STACK_TEXT:  
0177f778 6668be6c     0177f910 659b9a8f 00000000 game_sa!CBikeSA::RecalculateBikeHandling+0x3
0177f780 659b9a8f     00000000 4d3222bf 1c1ce638 game_sa!CBikeSA::SetBikeHandlingData+0x2c
0177f924 659c1316     3a0a5430 0177f96c 659b42af client!CClientVehicle::Create+0x103f
0177f930 659b42af     410dad00 1c516538 1c7bde38 client!CClientVehicle::StreamIn+0x46
0177f96c 659b364b     410dad00 08f71d58 1c051de8 client!CClientStreamer::Restream+0x41f
0177f988 6598130d     0177f998 363eedb8 c523f015 client!CClientStreamer::DoPulse+0x13b
0177f9a4 659708f4     4d322157 08f71d58 00000000 client!CClientManager::UpdateStreamers+0xad
0177faf8 6596e64b     4d3220db 01c640b0 01cdce50 client!CClientGame::DoPulses+0x1dc4
0177fb74 6959bcb3     01c640b0 09047d00 69554192 client!CClientGame::DoPulsePostFrame+0x6eb
0177fb80 69554192     3248811c 090b1fe8 09047d00 core!CModManager::DoPulsePostFrame+0x53
0177fc70 695fdf84     324881d8 090b1fe8 00000000 core!CCore::DoPostFramePulse+0x8b2
0177fcb4 69605170     09047d00 055ed6e8 007f99b0 core!CDirect3DEvents9::OnPresent+0x164
0177fd18 007f9b12     090b1fe8 00000000 00000000 core!CProxyDirect3DDevice9::Present+0x30
WARNING: Stack unwind information not available. Following frames may be wrong.
0177fd3c 007fb1c3     055ed6e8 003d050e 00000000 gta_sa+0x3f9b12
00000000 00000000     00000000 00000000 00000000 gta_sa+0x3fb1c3


FAULTING_SOURCE_LINE:  C:\TeamCity\buildAgent\work\675e5b8e8f135823\Client\game_sa\CBikeSA.cpp @ 56

IMAGE_NAME:  game_sa.dll

FAILURE_BUCKET_ID:  NULL_POINTER_READ_c0000005_game_sa.dll!CBikeSA::RecalculateBikeHandling
```
</details>

<details><summary>Dumptrace 2:</summary>

```
CONTEXT:  (.ecxr)
eax=5c582cb8 ebx=0177f924 ecx=00000000 edx=00000004 esi=00000007 edi=6d780c60
eip=5c4ebcd3 esp=0177f77c ebp=0177f780 iopl=0         nv up ei pl nz na pe nc
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00210206
game_sa!CBikeSA::RecalculateBikeHandling+0x3:
5c4ebcd3 8b01            mov     eax,dword ptr [ecx]  ds:002b:00000000=????????
Resetting default scope

EXCEPTION_RECORD:  (.exr -1)
ExceptionAddress: 5c4ebcd3 (game_sa!CBikeSA::RecalculateBikeHandling+0x00000003)
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000000
NumberParameters: 2
   Parameter[0]: 00000000
   Parameter[1]: 00000000
Attempt to read from address 00000000

PROCESS_NAME:  gta_sa.exe

READ_ADDRESS:  00000000 

ERROR_CODE: (NTSTATUS) 0xc0000005 - The instruction at 0x%p referenced memory at 0x%p. The memory could not be %s.

EXCEPTION_CODE_STR:  c0000005

EXCEPTION_PARAMETER1:  00000000

EXCEPTION_PARAMETER2:  00000000

STACK_TEXT:  
0177f778 5c4ebe6c     0177f910 5ba39a8f 00000000 game_sa!CBikeSA::RecalculateBikeHandling+0x3
0177f780 5ba39a8f     00000000 d272dd05 6d780c60 game_sa!CBikeSA::SetBikeHandlingData+0x2c
0177f924 5ba41316     6a6b0d88 0177f96c 5ba342af client!CClientVehicle::Create+0x103f
0177f930 5ba342af     41706400 17763408 17ce5c88 client!CClientVehicle::StreamIn+0x46
0177f96c 5ba3364b     41706400 0916fc30 17b571e0 client!CClientStreamer::Restream+0x41f
0177f988 5ba0130d     0177f998 37ddcb0c c5258345 client!CClientStreamer::DoPulse+0x13b
0177f9a4 5b9f08f4     d272deed 0916fc30 00000000 client!CClientManager::UpdateStreamers+0xad
0177faf8 5b9ee64b     d272df61 01ac33e8 01b3b500 client!CClientGame::DoPulses+0x1dc4
0177fb74 5d7cbcb3     01ac33e8 09230720 5d784192 client!CClientGame::DoPulsePostFrame+0x6eb
0177fb80 5d784192     cc340c27 0929ec20 09230720 core!CModManager::DoPulsePostFrame+0x53
0177fc70 5d82df84     cc340ce3 0929ec20 00000000 core!CCore::DoPostFramePulse+0x8b2
0177fcb4 5d835170     09230720 053ad6e8 007f99b0 core!CDirect3DEvents9::OnPresent+0x164
0177fd18 007f9b12     0929ec20 00000000 00000000 core!CProxyDirect3DDevice9::Present+0x30
WARNING: Stack unwind information not available. Following frames may be wrong.
0177fd3c 007fb1c3     053ad6e8 000505b2 00000000 gta_sa+0x3f9b12
00000000 00000000     00000000 00000000 00000000 gta_sa+0x3fb1c3


FAULTING_SOURCE_LINE:  C:\TeamCity\buildAgent\work\675e5b8e8f135823\Client\game_sa\CBikeSA.cpp @ 56
```
</details>

In the case of 1 dump, it happened a few minutes after joining a server and playing. In the other, it happened after more than 40 minutes of playing. Every time it's totally random.

**Version**
Client r20717

**Comment by Dutchman101** [comment]
cc @TheNormalnij as author of suspect change

**Comment by Dutchman101** [comment]
Hold on, i used the wrong symbols, which explains the total garbage stacktrace. I will edit main post

**Comment by TheNormalnij** [comment]
Thanks you for dumps. I will try to fix it now.

**Comment by TheNormalnij** [comment]
[https://upload.mtasa.com/u/782515977/dumps.zip_](https://upload.mtasa.com/u/782515977/dumps.zip_) private link

**Comment by Dutchman101** [comment]
Properly fixed in 6f4fff8

Do we keep the sanity checks added by saml1er? I guess so, extra safety and cannot hurt right?
